integration-tests/tests/src/tests/sync/partition-values.ts

```diff
@@ -0,0 +1,229 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import { ObjectId, UUID } from "bson";
+import { expect } from "chai";
+import Realm from "realm";
+import { authenticateUserBefore } from "../../hooks/authenticate-user-before";
+import { importAppBefore } from "../../hooks/import-app-before";
+const PvIntDog = {
+  name: "Dog",
+  primaryKey: "_id",
+  properties: {
+    _id: "objectId",
+    breed: "string?",
+    name: "string",
+    realm_id: "int?",
+  },
+};
+
+const PvStringDog = {
+  name: "Dog",
+  primaryKey: "_id",
+  properties: {
+    _id: "objectId",
+    breed: "string?",
+    name: "string",
+    realm_id: "string?",
+  },
+};
+
+const PvUuidDog = {
+  name: "Dog",
+  primaryKey: "_id",
+  properties: {
+    _id: "objectId",
+    breed: "string?",
+    name: "string",
+    realm_id: "uuid?",
+  },
+};
+
+const PvObjectIdDog = {
+  name: "Dog",
+  primaryKey: "_id",
+  properties: {
+    _id: "objectId",
+    breed: "string?",
+    name: "string",
+    realm_id: "objectId?",
+  },
+};
+
+const createConfig = (schema: Realm.ObjectSchema, user: Realm.User, partitionValue: any): Realm.Configuration => ({
+  schema: [schema],
+  sync: {
+    user,
+    partitionValue,
+    //@ts-expect-error SessionStopPolicy is a const enum which is removed at runtime.
+    _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+  },
+});
+
+describe("Partition-values", () => {
+  describe("integer", () => {
+    importAppBefore("pv-int-tests");
+    authenticateUserBefore();
+    it("works", async function (this: Mocha.Context & AppContext & UserContext) {
+      const realmConfigPrimary = createConfig(PvIntDog, this.user, 42);
+      const realmConfigSecondary = createConfig(PvIntDog, this.user, 43);
+
+      console.log(realmConfigPrimary);
+      // ensure clean starting point
+      Realm.deleteFile(realmConfigPrimary);
+
+      const realm1 = await Realm.open(realmConfigPrimary);
+      realm1.write(() => {
+        realm1.create("Dog", { _id: new ObjectId(), name: "King" });
```

Comment:
the `realm_id` is not set (which is the primary key).  I'm kind of surprised this works at all.
Just did a bit of reading.  I think in general, when the configuration is opened, it just applies a partition to the arbitrary value 42, and any data written there will be partitioned by this, without actually having to explicitly have it's object level field (in this case `realm_id`) to anything at all.
Seems like these tests don't really reflect how most users will use partition based sync, but we are trying to migrate everyone away from that anyway.
Nothing to do here, just was confused what was happening here.